File: src/main/java/com/ty/lab/service/LabstateService.java
package com.ty.lab.service;

import com.ty.lab.entity.PageResult;
import com.ty.lab.pojo.TbLabstate;

import java.util.List;

/**
 * 服务层接口
 *
 * @author Administrator
 */
public interface LabstateService {

    /**
     * 返回全部列表
     *
     * @return
     */
    public List<TbLabstate> findAll();

    /**
     * 返回分页列表
     *
     * @return
     */
    public PageResult findPage(int pageNum, int pageSize);

    /**
     * 增加
     */
    public void add(TbLabstate labstate);

    /**
     * 修改
     */
    public void update(TbLabstate labstate);

    /**
     * 根据ID获取实体
     *
     * @param id
     * @return
     */
    public TbLabstate findOne(Long id);

    /**
     * 批量删除
     *
     * @param ids
     */
    public void delete(Long[] ids);

    /**
     * 分页
     *
     * @param pageNum  当前页 码
     * @param pageSize 每页记录数
     * @return
     */
    public PageResult findPage(TbLabstate labstate, int pageNum, int pageSize);

}

File: src/main/resources/static/js/service/labstateService.js
//服务层
app.service('labstateService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../labstate/findAll');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../labstate/findPage?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../labstate/findOne?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../labstate/add',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../labstate/update',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../labstate/delete?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../labstate/search?page='+page+"&rows="+rows, searchEntity);
	}    	
});
